scripts/upgrade.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+SUFFIX=`date +%s | shasum | base64 | head -c 8 ; echo`
+KEY=
+UPGRADE_FROM=juno
+UPGRADE_TO=master
+
+usage()
+{
+    echo Usage: $0 "--key <key-name>" "--from <release-name>" "--to <release-name>" "--suffix <instance-name-suffix>" >&2
+    echo "Options:" >&2
+    echo "    -k | --key    The name of the public key to use in all the instances." >&2
+    echo "    -f | --from   The OpenStack release to start from. (defaults to juno)" >&2
+    echo "    -t | --to     The OpenStack release to upgrade to (defaults to master)." >&2
+    echo "    -s | --suffix A string to append to all instance names (defaults to a randomly generated string)." >&2
+    exit 1
+}
+
+while [[ $# > 0 ]]; do
+    opt="$1"
+    case $opt in
+        -k|--key)
+            KEY="$2"
+            shift
+            ;;
+        -f|--from)
+            UPGRADE_FROM="$2"
+            shift
+            ;;
+        -t|--to)
+            UPGRADE_TO="$2"
+            shift
+            ;;
+        -s|--suffix)
+            SUFFIX="$2"
+            shift
+            ;;
+        -h|--help)
+            usage
+            ;;
+        *)
+            break
+            ;;
+    esac
+    shift
+done
+
+if [ "$KEY" == "" ]; then
+    echo Error: Key must be specified.
+fi
+if [ "$UPGRADE_FROM" == "" ]; then
+    echo Error: Release to upgrade from must be specified.
+fi
+if [ "$UPGRADE_TO" == "" ]; then
+    echo Error: Release to upgrade to must be specified.
+fi
+if [ "$SUFFIX" == "" ]; then
+    echo Error: Suffix must be specified.
+fi
+if [ "$KEY" == "" -o "$UPGRADE_FROM" == "" -o "$UPGRADE_TO" == "" -o "$SUFFIX" == "" ]; then
+    usage
+fi
+
+START=$(date +%s);
+
+printf "Deploying RPC $UPGRADE_FROM release with stack name rpc-$SUFFIX.\n"
+heat stack-create -f rpc_multi_node.yml rpc-$SUFFIX -P "heat_git_version=$UPGRADE_FROM;os_ansible_git_version=$UPGRADE_FROM;deploy_monitoring=no;key_name=$KEY;cluster_prefix=rpc-$SUFFIX" -t 150 >/dev/null 2>&1
```

Comment:
perhaps only redirect stdout and leave stderr alone here.
would also be nice to pass --debug through to heat if the script is invoked with it.